hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -275,9 +275,12 @@ public HoodieCleanMetadata execute() {
             throw e;
           }
         }
-        table.getMetaClient().reloadActiveTimeline();
-        if (table.getMetaClient().getTableConfig().isMetadataTableAvailable()) {
-          table.getHoodieView().sync();
+        if (!pendingCleanInstants.get(pendingCleanInstants.size() - 1).equals(hoodieInstant)) {
```

Comment:
The existing timeline does not refresh, so how could this be true?

Reply:
The `hoodieInstant` is coming from the `pendingCleanInstants` list. This is just checking if we are on the last iteration.